scheduler/src/cook/mesos/util.clj

```diff
@@ -364,25 +422,31 @@
                                    (-> 1 t/hours t/in-millis)))
           expanded-end (Date. (+ (.getTime end)
                                  (-> 1 t/hours t/in-millis)))
+          entid-start (d/entid-at db :db.part/user expanded-start)
+          entid-end (d/entid-at db :db.part/user expanded-end)
           default-pool? (pool/default-pool? pool-name)
           pool-name' (or pool-name pool/nil-pool)
+          job-user-entid (d/entid db :job/user)
+          start-ms (.getTime start)
+          end-ms (.getTime end)
           jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
+          (->> (d/seek-datoms db :avet :job/user user entid-start)
```

Comment:
Why are we using `avet` here but `vaet` above? If this is intentional, I think we should explain why.

Reply:
Because the queries are different. Added a comment to the affect below.

Reply:
The comment below explains why we're using user here and state above, but it doesn't explain why we're using `avet` here but `vaet` above

Reply:
I don't have an answer other than 'because those are the indexes to optimially return results given the query type that we are performing.'

Reply:
Datomic doesn't do query optimization. We have to. Fortunately, the data shapes are uniform enough that we can do it statically and do a fair enough job, and those are the most sane static queries to use.